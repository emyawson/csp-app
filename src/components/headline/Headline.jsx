import React from 'react';
import PropTypes from 'prop-types';
import './Headline.scss';

const Headline = ({ title, description }) => (
  <div className="headline">
      <h1 className="headline__title" data-cytest="headline">{title}</h1>
      {description && <p className="headline__description">{description}</p>}
  </div>
);

Headline.propTypes = {
  title: PropTypes.string,
  description: PropTypes.string,
};

export default Headline;
