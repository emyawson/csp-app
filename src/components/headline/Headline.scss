@import '../../index';

.headline {
  color: $colour-neutral-grey-900;
  display: flex;
  flex-direction: column;
  justify-content: center;
  margin-top: 40px;
  margin-bottom: 40px;
  @include respond(tablet) {
    max-width: 466px;
    margin-top: 60px;
    width: auto;
  }
  @include respond(desktop) {
    max-width: 505px;
  }
}
.headline__btn {
  min-width: auto;
}
.headline__title {
  @extend %largest-titlebase;
}
.headline__description {
  @extend %description-copybase;
  color: $colour-neutral-grey-700;
  margin-top: 10px;
}