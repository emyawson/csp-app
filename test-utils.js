import React from 'react';
import { Provider } from 'react-redux';
import { createStore, combineReducers } from 'redux';
import { Router } from 'react-router-dom';
import { connectRouter } from 'connected-react-router';
import { render } from '@testing-library/react';
import { persistReducer } from 'redux-persist';
import storage from 'redux-persist/lib/storage';
import autoMergeLevel2 from 'redux-persist/lib/stateReconciler/autoMergeLevel2';
import { history } from './src/store';
import reducers from './src/store/reducer/index';

const createRootReducer = (reducerHistory) => combineReducers({
  router: connectRouter(reducerHistory),
  ...reducers,
});

// Setup redux persist
const persistConfig = {
  key: 'root',
  storage,
  stateReconciler: autoMergeLevel2,
};
const persistedReducer = persistReducer(
  persistConfig,
  createRootReducer(history),
);

const renderWithRouter = (ui, options = {}, initialState) => ({
  ...render(
      <Provider store={createStore(persistedReducer, initialState)}>
        <Router history={history}>{ui}</Router>
      </Provider>,
  ),
});

const AllTheProviders = () => {};
const customRender = (ui, options) => render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render, renderWithRouter };
