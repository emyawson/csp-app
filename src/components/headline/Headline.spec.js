import React from 'react';
// import { render, screen } from "@testing-library/react";
import Headline from './Headline.jsx';
// import { renderWithRouter } from "../test-utils";
import { renderWithRouter } from '../../../test-utils';

const defaultState = {};

const renderComponent = (state = {}) => {
  const combinedState = {
    ...defaultState,
    ...state,
  };
  return renderWithRouter(<Headline />, combinedState);
};

test('renders <Headline/>', () => {
  const { container } = renderComponent();
  expect(container).toMatchSnapshot();

  // const linkElement = screen.getByText(/learn react/i);
  // expect(linkElement).toBeInTheDocument();
});
