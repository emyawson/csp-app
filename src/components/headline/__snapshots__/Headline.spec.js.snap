// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <Headline/> 1`] = `
<div>
  <div
    class="headline"
  >
    <h1
      class="headline__title"
      data-cytest="headline"
    />
  </div>
</div>
`;
